### PR TITLE
[action][danger] Add fail-if-no-pr flag to Danger action

### PR DESCRIPTION
**Pull request by minuscorp**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
This PR adds the missing `fail-if-no-pr` flag of Danger action that was missing from the config.

### Description
Added the corresponding `ConfigItem` and append it to the command in the Danger call.

### Testing Steps
1. Checkout and install the following fastlane’s branch.
2. Use the Danger action and test that the `fail-if-no-pr` flag is added when configured like so.
3. 🚀

### CODE REVIEW

File: fastlane/lib/fastlane/actions/danger.rb [hunk 1]
```diff
@@ -99,7 +100,13 @@ def self.available_options
                                        env_name: "FL_DANGER_PR",
                                        description: "Run danger on a specific pull request. e.g. \"https://github.com/danger/danger/pull/518\"",
                                        is_string: true,
-                                       optional: true)
+                                       optional: true),
+          FastlaneCore::ConfigItem.new(key: :fail_if_no_pr,
+                                       env_name: "FL_DANGER_FAIL_IF_NO_PR",
+                                       description: "Fail Danger execution if no PR is found",
+                                       type: Boolean,
```
Comment (ainame) [inline fastlane/lib/fastlane/actions/danger.rb:107]: 😉

**Review by ainame — APPROVED** [review]
🥇

File: fastlane/lib/fastlane/actions/danger.rb [hunk 2]
```diff
@@ -99,7 +100,13 @@ def self.available_options
...
+                                       optional: true,
```
Comment (joshdholtz) [inline fastlane/lib/fastlane/actions/danger.rb:108]: Optional can be removed if there is a `default_value` for a `type: Boolean` 😊

**Review by joshdholtz — CHANGES_REQUESTED** [review]
Some small change otherwise 💯

**Review by joshdholtz — COMMENTED** [review]
Perfect! Thanks for adding this ❤️

**Review by joshdholtz — APPROVED** [review]

**Pull request merged**